File: Java/Array/Maximum Difference Between Adjacent Elements in a Circular Array.java
class Solution {
    public int maxAdjacentDistance(int[] nums) {
        int n=nums.length;
        int diff=Integer.MIN_VALUE;
        for(int i=0;i<n-1;i++){
            diff=Math.max(diff,Math.abs(nums[i]-nums[i+1]));
        }
        return Math.max(diff,Math.abs(nums[0]-nums[n-1]));
    }
}

File: CPP/Array/Maximum Difference Between Adjacent Elements in a Circular Array.cpp
class Solution {
public:
    int maxAdjacentDistance(vector<int>& nums) {
        int diff=INT_MIN;
        for(int i=0;i<nums.size()-1;i++){
            if(abs(nums[i]-nums[i+1])>diff){
                diff=abs(nums[i]-nums[i+1]);
            }
        }
        return max(diff,abs(nums[0]-nums[nums.size()-1]));
    }
};
